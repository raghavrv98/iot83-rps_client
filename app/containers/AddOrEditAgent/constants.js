/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
