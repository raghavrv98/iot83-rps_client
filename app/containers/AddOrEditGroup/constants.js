/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
