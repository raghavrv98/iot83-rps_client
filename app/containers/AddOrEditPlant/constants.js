/*
 *
 * AddOrEditPlant constants
 *
 */

export const DEFAULT_ACTION = "app/AddOrEditPlant/DEFAULT_ACTION";

export const CREATE_PLANT = "app/AddOrEditPlant/CREATE_PLANT";
export const CREATE_PLANT_SUCCESS = "app/AddOrEditPlant/CREATE_PLANT_SUCCESS";
export const CREATE_PLANT_FAILURE = "app/AddOrEditPlant/CREATE_PLANT_FAILURE";

export const GET_DETAILS = "app/AddOrEditPlant/GET_DETAILS";
export const GET_DETAILS_SUCCESS = "app/AddOrEditPlant/GET_DETAILS_SUCCESS";
export const GET_DETAILS_FAILURE = "app/AddOrEditPlant/GET_DETAILS_FAILURE";

export const UPLOAD_PLANT_IMAGE = "app/AddOrEditPlant/UPLOAD_PLANT_IMAGE";
export const UPLOAD_PLANT_IMAGE_SUCCESS = "app/AddOrEditPlant/UPLOAD_PLANT_IMAGE_SUCCESS";
export const UPLOAD_PLANT_IMAGE_FAILURE = "app/AddOrEditPlant/UPLOAD_PLANT_IMAGE_FAILURE";