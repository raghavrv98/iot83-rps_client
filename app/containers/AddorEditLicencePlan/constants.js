/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";