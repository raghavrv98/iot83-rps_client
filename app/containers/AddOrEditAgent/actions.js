/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
