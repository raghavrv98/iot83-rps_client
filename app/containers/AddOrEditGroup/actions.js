/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
