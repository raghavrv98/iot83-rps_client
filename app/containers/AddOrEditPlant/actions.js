
import { DEFAULT_ACTION,CREATE_PLANT,GET_DETAILS, UPLOAD_PLANT_IMAGE } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function createPlantHandler(payload,id) {
  return {
    type: CREATE_PLANT,
    payload,
    id
  };
}

export function fetchPlantDetails(id) {
  return {
    type: GET_DETAILS,
    id
  };
}

export function uploadPlantImage(image) {
  return {
    type: UPLOAD_PLANT_IMAGE,
    image,
  };
}