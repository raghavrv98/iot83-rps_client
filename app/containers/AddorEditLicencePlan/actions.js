/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}