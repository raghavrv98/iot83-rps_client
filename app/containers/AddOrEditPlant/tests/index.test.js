import React from 'react';
import { shallow } from 'enzyme';
import history from 'utils/history';
import { fromJS } from 'immutable';
import { createPlantHandler, fetchPlantDetails, uploadPlantImage } from '../actions';
import { AddOrEditPlant, mapDispatchToProps} from '../index';
import { CREATE_PLANT,GET_DETAILS, UPLOAD_PLANT_IMAGE } from '../constants';

let match = {
  params:{
    id:"demoId",
  }
}

const list = {
  uploadedImage: "",
  mobilePhoneError: "",
  payload: {
    name: "",
    ownerName: "",
    ownerEmail: "",
    ownerPhone: "",
    plantImage: null,
  },
  phone: ""
};

const createPlantSuccess = "this is submit success";
const createPlantFailure = "this is submit error";
const getDetailsSuccess = "this is plant details success";
const getDetailsFailure = "this is plant details Error"; 
const plantImageUplaod = "this is image success";
const plantImageUplaodFailure = "this is image error";
describe('< AddOrEditPlant />', () => {

  it('match snapshots', () => {
    const wrap = shallow(<AddOrEditPlant match={match} createPlantHandler={createPlantHandler} fetchPlantDetails={fetchPlantDetails} uploadPlantImage={uploadPlantImage} />)
    expect(wrap).toMatchSnapshot()
  })

it("clicking back button trigger onClick event", async () => {
    let expectedCount = history.length + 1;
    const wrapper = shallow(<AddOrEditPlant history={history} match={match} fetchPlantDetails={fetchPlantDetails} />);
    let backbutton = wrapper.find("button.btn-transparent");
    expect(backbutton.length).toBe(1);
    backbutton.simulate("click");
    expect(history.length + 1).toEqual(expectedCount);
    expect(history.location.pathname).toEqual("/");
});
  

//   it("clicking submit button trigger onClick event", async () => {
//     const createPlantHandler = jest.fn();
//     const wrapper = shallow(<AddOrEditPlant history={history} match={match} createPlantHandler={createPlantHandler} fetchPlantDetails={fetchPlantDetails} />);
//     wrapper.setState({ uploadPlantImage: 'uploadPlantImage' });
//     const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'submitHandler');
//     wrapper.instance().forceUpdate(); 
//     let addOrEditForm = wrapper.find("form.contentForm");
//     expect(addOrEditForm.length).toBe(1);
//     addOrEditForm.simulate('submit', { preventDefault() { } });
//     expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
//   });

//  it("clicking onChange button trigger onClick event", async () => {
//   const createPlantHandler = jest.fn();
//   const wrapper = shallow(<AddOrEditPlant history={history} match={match} createPlantHandler={createPlantHandler} fetchPlantDetails={fetchPlantDetails} />);
//   const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'nameChangeHandler');
//   wrapper.instance().forceUpdate(); 
//   wrapper.find('[id="name"]').simulate('change',{target: {id: "name", value: "name"}});
//   expect(wrapper.find('[id="name"]').props().value).toEqual("name");
//   expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
// });

  it("test component will recieve props", async () => {
    spyOn( AddOrEditPlant.prototype, "componentWillReceiveProps").and.callThrough();
    const wrapper = shallow(<AddOrEditPlant history={history} match={match} fetchPlantDetails={fetchPlantDetails} uploadPlantImage={uploadPlantImage} createPlantHandler={createPlantHandler} createPlantSuccess={undefined} createPlantFailure={undefined} getDetailsSuccess={undefined} getDetailsFailure={undefined} plantImageUplaod={undefined} plantImageUplaodFailure={undefined} />);
    wrapper.setProps({ createPlantSuccess, createPlantFailure, getDetailsSuccess, getDetailsFailure, plantImageUplaod, plantImageUplaodFailure });
    wrapper.setState({resList:list.payload});
    expect( AddOrEditPlant.prototype.componentWillReceiveProps ).toHaveBeenCalledTimes(1);
  });

  it("test componentDidMount", async () => {
    const fetchPlantDetails=jest.fn();
    spyOn(AddOrEditPlant.prototype, "componentDidMount").and.callThrough();
    const wrapper = shallow(<AddOrEditPlant history={history} match={match} fetchPlantDetails={fetchPlantDetails} />);
    wrapper.setState({ })
    expect(AddOrEditPlant.prototype.componentDidMount).toHaveBeenCalledTimes(1);
    // expect(fetchPlantDetails).toHaveBeenCalledTimes(1);
  });



  it("Function testing on mapDispatchToProps", () => {
    const dispatch = jest.fn();
    const payload = "demo";
    const id = "demoId";
    const image="image";
  
    mapDispatchToProps(dispatch).createPlantHandler(payload, id);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: CREATE_PLANT,
      payload: "demo",
      id: "demoId"
    });
    mapDispatchToProps(dispatch).fetchPlantDetails( id );
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: GET_DETAILS,
      id: "demoId",
    });
    mapDispatchToProps(dispatch).uploadPlantImage(image);
    expect(dispatch.mock.calls[2][0]).toEqual({
      type: UPLOAD_PLANT_IMAGE,
      image:"image",
    });
  });
});