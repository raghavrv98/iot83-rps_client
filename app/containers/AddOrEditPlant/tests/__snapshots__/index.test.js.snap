// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< AddOrEditPlant /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      AddOrEditPlant
    </title>
    <meta
      content="Description of AddOrEditPlant"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p
        className="pd-l-30"
      >
        <span
          className="cursor-pointer"
          onClick={[Function]}
        >
          <button
            className="btn btn-transparent"
          >
            <i
              className="far fa-long-arrow-left"
            />
          </button>
          <FormattedMessage
            defaultMessage="Manage Plants"
            id="app.common.messages.managePlant"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </span>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="Edit Plant"
          id="app.containers.AddOrEditPlant.editPlant"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonNestedFormUpload"
  />
</div>
`;
